File: 13-01-2025_Container_with_most_Water.cpp
class Solution {

  public:
    int maxWater(vector<int> &arr) {
        // code here
        if(arr.size()<2)
        {
            return 0;
        }
        int left = 0;
        int right = arr.size()-1;
        int max = 0;
        while(left < right)
        {
            int temp;
            if(arr[left]<arr[right])
            {
                temp = arr[left];
            }
            else
            {
                temp = arr[right];
            }
            temp *= (right-left);
            max = max > temp? max : temp;
            if(arr[left]<arr[right])
            {
                left++;
            }
            else
            {
                right--;
            }
        }
        return max;
    }
};

File: 13-01-2025_Container_with_most_Water.py
class Solution:
    def maxWater(self, arr):
        if len(arr) < 2:
            return 0
        left = 0
        right = len(arr) - 1
        max_val = 0

        while left < right:
            temp = min(arr[left], arr[right]) * (right - left)
            max_val = max(max_val, temp)

            if arr[left] < arr[right]:
                left += 1
            else:
                right -= 1

        return max_val
